src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,117 @@ public static string GetMemberName(ISymbol symbol)
+                || access == Accessibility.ProtectedAndInternal
+                || access == Accessibility.NotApplicable;
+        }
+
+        public static bool SymbolIsProtectedInSealed(ISymbol symbol)
+        {
+            var containgType = symbol.ContainingType;
+            if (containgType != null && containgType.IsSealed)
+            {
+                var access = symbol.DeclaredAccessibility;
+                return access == Accessibility.Protected
+                    || access == Accessibility.ProtectedOrInternal;
+            }
+
+            return false;
+        }
+
+        public static string GetMeaningfulParentName(SyntaxNode cur, SemanticModel model)
+        {
+            while (cur.Parent != null)
+            {
+                var pNode = cur.Parent;
+                ISymbol sym = pNode.GetDeclaredOrReferencedSymbol(model);
+
+                if (sym != null &&
+                    !string.IsNullOrEmpty(sym.Name)
+                    && (
+                        sym.Kind == SymbolKind.Method ||
+                        sym.Kind == SymbolKind.NamedType
+                       )
+                )
+                {
+                    return sym.Name;
+                }
+
+                cur = pNode;
+            }
+
+            return String.Empty;
+        }
+
+/// <summary>
+/// Determine wether a type (given by name) is actually declared in the expected assembly (also given by name)
+/// </summary>               
+/// <remarks>
+/// This can be used to decide wether we are referencing the expected framework for a given type. 
+/// For example, System.String exists in mscorlib for .NET Framework and System.Runtime for other framework (e.g. .NET Core). 
+/// </remarks>
+public static bool? IsTypeDeclaredInExpectedAssembly(Compilation compilation, string typeName, string assemblyName)
+        {
+            if (compilation == null)
+            {
+                return null;
+            }
+            var typeSymbol = compilation.GetTypeByMetadataName(typeName);
+            return typeSymbol?.ContainingAssembly.Identity.Name.Equals(assemblyName, StringComparison.Ordinal);
+        }
+        
+        /// <summary>
+        /// Gets the version of the target .NET framework of the compilation.
+        /// </summary>                          
+        /// <returns>
+        /// Null if the target framenwork is not .NET Framework.
+        /// </returns>
+        /// <remarks>
+        /// This method returns the assembly version of mscorlib for .NET Framework prior version 4.0, 
+        /// i.e. for .NET framework 3.5, the returned version would be 2.0.0.0.
+        /// For .NET Framework 4.X, this method returns the actual framework version instead of assembly verison of mscorlib,
+        /// i.e. for .NET framework 4.5.2, this method return 4.5.2 instead of 4.0.0.0.
+        /// </remarks>
+        public static Version GetDotNetFrameworkVersion(Compilation compilation)
```

Comment:
Consider move this method and `IsTypeDeclaredInExpectedAssembly` to Desktop.Analyzer assembly, it's unlikely to be used by other analyzers.